src/backend/src/lib.rs

```diff
@@ -154,36 +155,63 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+fn update_config(arg: InitArg) {
+    mutate_state(|state| {
+        let ic_root_key_raw = match arg
+            .ic_root_key_der
+            .map(|der_key| extract_raw_root_pk_from_der(&der_key))
+        {
+            Some(Ok(root_key)) => Some(root_key),
+            Some(Err(_)) => None,
+            None => match extract_raw_root_pk_from_der(IC_ROOT_PK_DER) {
+                Ok(root_key) => Some(root_key),
+                Err(_) => None,
+            },
+        };
+        state
+            .config
+            .set(Some(Candid(Config {
+                ecdsa_key_name: arg.ecdsa_key_name,
```

Comment:
Not sure but you might be able to destruct the arg here to be less version.

```
Config {
   ic_root_key_raw,
   ..arg
}
```

But agin, not sure.

Reply:
Sure!